[docs] Cover webpack 4 support in migration guide (#12710)
mui/mui-x: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md

If you need support for IE 11, you will need to keep using the latest version of the `v6` release.
:::

### Drop Webpack 4 support

Dropping old browsers support also means that we no longer transpile some features that are natively supported by modern browsers – like [Nullish Coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) and [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

These features are not supported by Webpack 4, so if you are using Webpack 4, you will need to transpile these features yourself or upgrade to Webpack 5.

Here is an example of how you can transpile these features on Webpack 4 using the `@babel/preset-env` plugin:

```diff
 // webpack.config.js

 module.exports = (env) => ({
   // ...
   module: {
     rules: [
       {
         test: /\.[jt]sx?$/,
-        exclude: /node_modules/,
+        exclude: [
+          {
+            test: path.resolve(__dirname, 'node_modules'),
+            exclude: [

> +              path.resolve(__dirname, 'node_modules/@mui/x-data-grid'),

Reviewer: I would be more explicit here. I guess nobody wants to do this to the license and the community package, they always want either to add some commercial package or skip the license.

I see 4 potential approaches:

- just add a sentence (the one I wrote is quite bad) 

  ```suggestion
  +              // Add all folder of each grid packages you are using.
  +              path.resolve(__dirname, 'node_modules/@mui/x-data-grid'),
  ```

- list the 3 cases (`@mui/x-data-grid` | `@mui/x-data-grid` + `@mui/x-data-grid-pro` + `@mui/x-license` | `@mui/x-data-grid` + `@mui/x-data-grid-pro` + `@mui/x-data-grid-premium | `@mui/x-license`

- list all the packages but add a small `// (only needed for pro and premium users)` and `// (only needed for premium users)` after the `path.resolve(...)` of each commercial package.

- list all the packages even if they are not useful, I guess they do no harm and make sure people don't have nasty bugs if they buy a license in the future.

Author: If they only use the community package, the license package won't be installed, so there's no harm in having it there.
I would rather keep the universal snippet that would work for every plan.

Reviewer: > I would rather keep the universal snippet that would work for every plan.

But do we agree that your snippet do not include the pro and premium packages and that can be problematic?
Or does `node_modules/@mui/x-data-grid` also match `node_modules/@mui/x-data-grid-pro`? In which case maybe just a comment saying that it does would be interesting.

Basically, just by reading the code snippet, the first thing I'd like to do as a pro user is to add `-pro` on the grid package line, and that is wrong because it will stop transpiling the community package, which is always needed.

Author: > But do we agree that your snippet do not include the pro and premium packages and that can be problematic?
Or does node_modules/@mui/x-data-grid also match node_modules/@mui/x-data-grid-pro?

I tested the Pro and Premium packages with this snippet and it worked.
I will check again to make sure it does work, and add a comment if that's the case 👍🏻

If you need support for IE 11, you will need to keep using the latest version of the `v6` release.
:::

> ### Drop webpack 4 support

Reviewer 2: However, I'm not sure that this is the most appropriate heading for this section. 🙈 
Technically, we only made our bundles slightly more modern, the problems with Webpack can be overcome by using transformers. 🤔 
```suggestion
### Drop Webpack 4 support
```

Author: I think it's OK.
Alternatively, I can call it "Drop out of the box Webpack 4 support", but it's too verbose 😅

Dropping old browsers support also means that we no longer transpile some features that are natively supported by modern browsers – like [Nullish Coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) and [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

> These features are not supported by webpack 4, so if you are using webpack 4, you will need to transpile these features yourself or upgrade to webpack 5.

Reviewer 2: ```suggestion
These features are not supported by Webpack 4, so if you are using Webpack 4, you will need to transpile these features yourself or upgrade to Webpack 5.
```

> Here is an example of how you can transpile these features for webpack 4 using `@babel/preset-env` plugin:

Reviewer 2: ```suggestion
Here is an example of how you can transpile these features on Webpack 4 using the `@babel/preset-env` plugin:
```
